Add Frontend Web Dev with Blazor Series (#4)
katiesavage/beginner-series: Frontend Web with Blazor/README.md

# Frontend Web Development with Blazor for Beginners

.NET makes frontend web development easy with Blazor, a modern frontend web framework based on HTML, CSS, and C#. In this video series we'll learn the basics of frontend web development with Blazor, including how to set up your development environment, create a new project, run & debug, and work with Blazor components.

## Video Series Links

> Full Series:
> YouTube link: https://aka.ms/dotnet/beginnervideos/youtube/blazor 
> 
> Learn link: https://aka.ms/dotnet/beginnervideos/learn/blazor

Reviewer: ```suggestion
Full Series:
- YouTube link: https://aka.ms/dotnet/beginnervideos/youtube/blazor 
- Learn link: https://aka.ms/dotnet/beginnervideos/learn/blazor
```

Individual Video Links:
1. [What is Blazor?]()
1. [Build and run your first Blazor web app]()
1. [Introducing Blazor Components]()
1. [Pages, Routing, and Layouts in Blazor]()
1. [Blazor Web App Project Structure]()
1. [Handling UI Events in Blazor Web Apps]()
1. [Data Binding in Blazor Web Apps]()
1. [Blazor Component Render Modes]()
1. [Work with Data in Blazor]()
1. [Forms & Validation in Blazor Web Apps]()

## Overview

| **Goal**              | Build your first Blazor web application                                    |
| ----------------------------- | --------------------------------------------------------------------- |
| **What will you learn**       | Introduction to Frontend Web Development with .NET using Blazor                                       |

> | **What you'll need**          | [VS Code](code.visualstudio.com) or [Visual Studio](https://visualstudio.microsoft.com/downloads/), [C# Dev Kit](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csdevkit) extension if you use VS Code, [.NET SDK](https://dotnet.microsoft.com/download)|

Reviewer: ```suggestion
| **What you'll need**          | [Visual Studio Code](code.visualstudio.com) with the [C# Dev Kit](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csdevkit) extension and the [.NET SDK](https://dotnet.microsoft.com/download)|, or [Visual Studio](https://visualstudio.microsoft.com/downloads/) with the "ASP.NET and web development" workload installed.
```

| **Number of videos**          | 10                                                               |
| **Follow Along** | [sample-code](sample-code)                        |
                         

## Video

Full Series:
YouTube link: https://aka.ms/dotnet/beginnervideos/youtube/blazor 

Learn link: https://aka.ms/dotnet/beginnervideos/learn/blazor

## Video 1: What is Blazor?

> Welcome to the Frontend Web Development for Beginners! In this expisode you'll learn about what Blazor is and how it's used from frontend web development with .NET.

Reviewer: ```suggestion
Welcome to the Frontend Web Development for Beginners! In this episode you'll learn about what Blazor is and how it's used for frontend web development with .NET.
```

[What is Blazor?]()
 
## Video 2: Build and run your first Blazor web app

> It's time to get you development environment set up to build your first Blazor web app.

Reviewer: ```suggestion
It's time to get your development environment set up to build your first Blazor web app.
```

 
[Build and run your first Blazor web app]()

## Video 3: Introducing Blazor Components

> This episode is all about Blazor components. You'll learn how to create components, how to use them, and how pass data to them.

Reviewer: ```suggestion
This episode is all about Blazor components. You'll learn how to create components, how to use them, and how to pass data to them.
```

 
[Introducing Blazor Components]()

## Video 4: Pages, Routing, and Layouts in Blazor
In this episode you'll learn how to route page navigations to specific components and how to set up a layout for your pages.
 
[Pages, Routing, and Layouts in Blazor]()
 
## Video 5: Blazor Web App Project Structure

> Let's take a look at all the files that make up a Blazor web app project and understand what they do.

Reviewer: ```suggestion
Let's take a look at all the files that make up a Blazor Web App project and understand what they do.
```

 
[Blazor Web App Project Structure]()

## Video 6: Handling UI Events in Blazor Web Apps
In this episode you'll learn how to handle all kinds of UI events from your components. We'll also look at handling component child content, working with render fragments, and handling additional component attributes.

[Handling UI Events in Blazor Web Apps]()
 
## Video 7: Data Binding in Blazor Web Apps

> Learn how to bind data to your UI elements and components using Blazor's powerful data binding capabilities. We'll also take everything we've learned so far and build a simple to do list app.

Reviewer: ```suggestion
Learn how to bind data to your UI elements and components using Blazor's powerful data binding capabilities. We'll also take everything we've learned so far and build a simple to-do list app.
```
